assets/scripts/dialogs/AnalyticsDialog.jsx

```diff
@@ -60,13 +63,26 @@ const avgCapacityAscending = (a, b) => {
 }
 
 function AnalyticsDialog (props) {
+  const dispatch = useDispatch()
   const street = useSelector((state) => state.street)
+  console.log({ street, props })
   const locale = useSelector((state) => state.locale.locale)
 
   useEffect(() => {
     trackEvent('Interaction', 'Open analytics dialog box', null, null, false)
   }, [])
 
+  const [isToggleDisabled, setIsToggleDisabled] = useState(false)
+  const [isVisible, setIsVisible] = useState(street.showAnalytics)
+  const toggleIsVisible = () => {
+    setIsToggleDisabled(true)
+    setIsVisible(!isVisible)
+    dispatch(updateAnalytics(!isVisible))
+    saveStreetToServer(true)
+    setIsToggleDisabled(false)
```

Comment:
I do have one further comment / question about the use of the disabled state itself. Do we need it? It seems like the intent is to disable the UI from receiving additional input until after the updates have occurred, but all of the function calls here are asynchronous. I haven't tested this locally, but my reading of this code is that the disabled toggle happens so quickly that the user would never see it. Let me know if this is false.

One other reason why we might not need it is:
1) We don't do this on any of the other settings that kick off a server request, so unless we know we're trying to introduce this codebase-wide, we shouldn't have multiple ways of doing this
2) Saving street to server (and several other server requests that might happen in quick succession) should already be debounced by the `utils/fetch_nonblocking.js` module. In other words, the way we have designed this, it's not the responsibility of the UI to ensure that a save has happened before allowing a user to click the checkbox again. It should be the responsibility of our saving mechanism to handle rapid succession of input.

Reply:
As a follow up note, I think ultimately we want a dispatch to store to kick off the server request -- it doesn't need to happen here.